### PR TITLE
Updated project build and dependencies

### PR DESCRIPTION
**Pull request by blootsvoets**
- MP Auth 0.8.2 -> 2.0.0
- Kotlin 1.3 -> 1.8
- Java 8 -> 17
- Gradle 6 -> 8
- Github Actions latest versions (removing deprecated Node 12 calls)
- Spring 5 -> 6
- Added several build features
  - ktlint for linting (and fixed listing issues)
  - dependencyUpdates for easier updating
  - dokka Javadoc-compatible documentation
  - put all plugin and dependency versions in `gradle.properties`

### CODE REVIEW

**Review by yatharthranjan — COMMENTED** [review]
Thanks for the substantial update. 

I am wondering there are no other changes required for MP 2.0? Like organisation being added to the authorisation?

**Comment by blootsvoets** [comment]
Yes, I am working on a radar-auth overhaul which would be more substantial and makes it easier to do full organisation authorisation. Since you only use radar-spring in radar-appserver at the moment, which is just used by participants, I didn't add the logic to map projects to organisations and vice-versa. The participant authorization has not changed.

**Comment by yatharthranjan** [comment]
> Yes, I am working on a radar-auth overhaul which would be more substantial and makes it easier to do full organisation authorisation. Since you only use radar-spring in radar-appserver at the moment, which is just used by participants, I didn't add the logic to map projects to organisations and vice-versa. The participant authorization has not changed.

ok great, makes sense. Thanks.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -17,24 +17,14 @@ jobs:
...
-          # Cache gradle directories
-          path: |
-            ~/.gradle/caches/jars-3
-            ~/.gradle/caches/modules-2/files-2.1/
-            ~/.gradle/caches/modules-2/metadata-2.96/
-            ~/.gradle/native
-            ~/.gradle/wrapper
-          # Key for restoring and saving the cache
-          key: ${{ runner.os }}-gradle
+      - uses: gradle/gradle-build-action@v2
```
Comment (yatharthranjan) [inline .github/workflows/main.yml:27]: is linting automatically checked when using `./gradlew check`?

**Review by yatharthranjan — APPROVED** [review]
LGTM, thanks. 

Just a minor question.

File: .github/workflows/main.yml (hunk 1 above)
Comment (blootsvoets) [reply .github/workflows/main.yml:27]: Yes indeed. Explicitly check with `ktlintCheck` and automatically fix with `ktlintFormat`.

**Pull request merged**